handler.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+var AWS = require('aws-sdk');
+var attr = require('dynamodb-data-types').AttributeValue;
+var craigslistApartments = require('./craigslist-apartments');
+var filters = require('./filters');
+
+
+var ddb = new AWS.DynamoDB({apiVersion: '2012-08-10'});
+
+module.exports.hello = (event, context) => {
+  const filter = new filters(event.filters.postalCode, event.filters.kmFromPostalCode, event.filters.minPrice, event.filters.maxPrice, event.filters.minSqft);
+  var apartmentsPromise = craigslistApartments.getListOfRentals(filter);
+  return apartmentsPromise.then((result) => {
+    return result;
+  })
+  .then((apartmentListings) => {
+    return storeInDB(ddb, apartmentListings);
+  })
+  .then((todaysListings) => {
+    var yesterdayListingsPromise = getYesterdaysListings(ddb);
+    return yesterdayListingsPromise.then((result) => {
+      return result;
+    })
+    .then((yesterdayListings) => {
+      var apartmentListings = {
+        todaysListings: todaysListings,
+        yesterdayListings: yesterdayListings
+      }
+      return apartmentListings;
+    })
+    
+  })
+  .then((apartmentListings) => {
+    var apartments = findNewApartments(apartmentListings);
+    var removed = findRemovedApartments(apartmentListings);
+    var message = createdMessage(apartments, removed);
+    return sendEmail(message);
+  })
+  .then(() => {
+    return {
+      statusCode: 200,
+      body: JSON.stringify({ message: "Lambda executed with success" })
+    }
+  })
+}
+
+function findNewApartments(apartmentListings) {
+  var apartments = {
+    newApartments: [],
+    oldApartments: []
+  };
+  apartmentListings.todaysListings.forEach(function(element) {
+    var elementPos = apartmentListings.yesterdayListings.listings.map(function(x) {return x.id; }).indexOf(element.id); 
```

Comment:
nitpick: For this case, where we'll have a small list of elements, this won't have much effect, but ideally you'd come up with a more efficient solution (for instance, turn the lists into maps of id -> element).